elide-model-config/src/main/java/com/yahoo/elide/modelconfig/validator/DynamicConfigValidator.java

```diff
@@ -252,56 +251,156 @@ private static void validateInheritance(ElideTableConfig tables, Table table, Se
     }
 
     private void populateInheritance(ElideTableConfig elideTableConfig) {
-        for (Table table : elideTableConfig.getTables()) {
-            if (table.hasParent()) {
-                Map<String, Measure> measures = getInheritedMeasures(table, new HashMap<String, Measure>());
-                table.setMeasures(new ArrayList<Measure>(measures.values()));
+        //ensures validation is run before populate always.
+        validateInheritance(this.elideTableConfig);
 
-                Map<String, Dimension> dimensions = getInheritedDimensions(table, new HashMap<String, Dimension>());
-                table.setDimensions(new ArrayList<Dimension>(dimensions.values()));
+        Set<Table> processed = new HashSet<Table>();
+        elideTableConfig.getTables().stream().forEach(table -> {
+            populateInheritance(table, processed);
+        });
+    }
 
-                Map<String, Join> joins = getInheritedJoins(table, new HashMap<String, Join>());
-                table.setJoins(new ArrayList<Join>(joins.values()));
+    private void populateInheritance(Table table, Set<Table> processed) {
+        if (table.hasParent() && !processed.contains(table)) {
+            Table parent = table.getParent(this.elideTableConfig);
+            // If parent also extends, ensure parent is processed first.
+            if (parent.hasParent()) {
+                populateInheritance(parent, processed);
             }
+
+            Map<String, Measure> measures = getInheritedMeasures(table, new HashMap<String, Measure>());
```

Comment:
shouldn't this be called on the parent rather than table?  Similar comment for other calls below.

Reply:
We first retrieve the measure list from current table and populate in the new HashMap in the `getInheritedMeasures` method. And in the same method we call it for `Parent` class too. Refer L-315.

Reply:
But we shouldn't need to _add_ the measures for the current table (as they are already defined in the table).  We only need to add the inherited measures.  This reads like we are duplicating what's already defined in the current table.